x-pack/plugins/enterprise_search/public/applications/app_search/components/create_engine/constants.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+
+export const CREATE_ENGINE_TITLE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.title',
+  {
+    defaultMessage: 'Create an Engine',
+  }
+);
+
+export const CREATE_ENGINE_FORM_TITLE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.title',
+  {
+    defaultMessage: 'Name your Engine',
+  }
+);
+
+export const CREATE_ENGINE_FORM_ENGINE_NAME_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.engineName.label',
+  {
+    defaultMessage: 'Engine Name',
+  }
+);
+
+export const ALLOWED_CHARS_NOTE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.engineName.allowedCharactersNote',
+  {
+    defaultMessage: 'Engine names can only contain lowercase letters, numbers, and hyphens',
+  }
+);
+
+export const SANITIZED_NAME_NOTE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.engineName.sanitizedNameNote',
+  {
+    defaultMessage: 'Your engine will be named',
+  }
+);
+
+export const CREATE_ENGINE_FORM_ENGINE_NAME_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.engineName.placeholder',
+  {
+    defaultMessage: 'i.e., my-search-engine',
+  }
+);
+
+export const CREATE_ENGINE_FORM_ENGINE_LANGUAGE_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.engineLanguage.label',
+  {
+    defaultMessage: 'Engine Language',
+  }
+);
+
+export const CREATE_ENGINE_FORM_SUBMIT_BUTTON_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.form.submitButton.label',
+  {
+    defaultMessage: 'Create Engine',
+  }
+);
+
+export const CREATE_ENGINE_SUCCESS_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.createEngine.successMessage',
+  {
+    defaultMessage: 'Successfully created engine.',
+  }
+);
+
+export const SUPPORTED_LANGUAGES = [
+  {
+    value: 'Universal',
+    text: 'Universal',
```

Comment:
😬 Super sorry about this, but I think we have to i18n all of these `text` values. I think the message type Kibana uses for these is `DropDownOptionLabel`

Reply:
Also, I'm now happy we decided to move these options to the front-end - we wouldn't have gotten i18n localization if we'd kept the values coming in from the server 🎉